### PR TITLE
Sc 3312/amazon s3 remove file s

### PR DESCRIPTION
**Pull request by Shamwayne**

### CODE REVIEW

**Comment by smjohnsonShipyard** [comment]
image
Not sure if this is fixable on your side, but I put a file name that doesn't exist in S3 and it still says that the file can be deleted

**Comment by smjohnsonShipyard** [comment]
Make sure you update the requirements.txt for both of these as well.

File: amazons3_blueprints/remove_files.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import os
+import boto3
+import botocore
+from botocore.client import Config
+import re
+import argparse
+import glob
+from ast import literal_eval
```
Comment (ericelsken) [inline amazons3_blueprints/remove_files.py:8]: This isn't being used and can be removed?

File: amazons3_blueprints/remove_files.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    if source_file_name_match_type == 'regex_match':
+        file_names = s3_list_files(
+            s3_connection, bucket_name, source_folder_name)
+        matching_file_names = shipyard.files.find_all_file_matches(
+            file_names, re.compile(source_file_name))
+        num_matches = len(matching_file_names)
+
+        if num_matches == 0:
+            print(f'No matches found for regex {source_file_name}')
+            sys.exit(1)
```
Comment (ericelsken) [inline amazons3_blueprints/remove_files.py:132]: Do we want an non-succesful exit code if no files are found, @smjohnsonShipyard ?

File: amazons3_blueprints/remove_files.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        else:
+            print(f'{num_matches} files found. Preparing to upload...')
```
Comment (ericelsken) [inline amazons3_blueprints/remove_files.py:134]: upload -> remove.

**Review by ericelsken — CHANGES_REQUESTED** [review]

File: amazons3_blueprints/remove_files.py (hunk 2 above)
Comment (smjohnsonShipyard) [reply amazons3_blueprints/remove_files.py:132]: I believe that would make sense. [](edited-with-Axolo)

**Pull request merged**